src/lib/base.jl

```diff
@@ -29,7 +29,9 @@ function accum(a::AbstractDict, b::AbstractDict)
 end
 
 @adjoint function getindex(d::AbstractDict, k)
-  d[k], function (Δ)
+  val = d[k]
+  val, function (Δ)
```

Comment:
```suggestion
  val, function Dict_getindex_pullback(Δ)
```

Reply:
Sure! Should I add it to the `lib.jl` test set in the file `test/lib/lib.jl`? Or, mirroring, the source, should I make a new `test/lib/base.jl` file and test set?

Reply:
Regarding naming the pullback: none of the other pullbacks created in `base.jl` seem to have names. Returning an anonymous function seems like the preferred style.

Reply:
The latter, ideally. Currently a bunch of tests are all jammed into https://github.com/FluxML/Zygote.jl/blob/master/test/features.jl (including the current `getindex(::Dict)` ones, which you need not worry about for this PR), and we might as well take this opportunity to start breaking that file up into a saner file structure :)

Reply:
Ideally we would like to turn all the anonymous functions into named ones and also move them to ChainRules' rrules

Reply:
What Carlo said, but that will be a longer term project and `getindex` is a little special. Were this most other functions, I would've recommended rewriting it as a `rrule` off the bat.

> none of the other pullbacks created in base.jl seem to have names. Returning an anonymous function seems like the preferred style.

Yes, this is a legacy code style and has caused no shortage of headaches while debugging. We plan on either removing them or giving them a name sooner than later, but in the meantime we can at least advocate for new pullbacks to have descriptive names.